crates/net/network/src/manager.rs

```diff
@@ -510,6 +510,17 @@ where
             NetworkHandleMessage::DisconnectPeer(peer_id, reason) => {
                 self.swarm.sessions_mut().disconnect(peer_id, reason);
             }
+            NetworkHandleMessage::Shutdown => {
+                // Set connection status to `Shutdown`. This will prevent the node to accept
+                // new incoming connections as well as sending connection requests to newly
+                // discovered nodes.
+                self.swarm.set_shutdown_connection_state();
+
+                // drop pending connections
```

Comment:
we can do this similarly to how we disconnect active sessions by looping over pending session and sending a disconnect message:

we can finally use this:

https://github.com/paradigmxyz/reth/blob/d08aa94f5d7cee743ab05526dd562fcb8e8c3d42/crates/net/network/src/session/handle.rs#L25-L27

unsure if we can send the disconnect reason here though